src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java

```diff
@@ -661,6 +661,29 @@ protected void addBeanProps(DeserializationContext ctxt,
         }
     }
 
+    // since 2.18
+    private SettableAnyProperty _resolveAnySetter(DeserializationContext ctxt,
+              BeanDescription beanDesc, SettableBeanProperty[] creatorProps)
+            throws JsonMappingException
+    {
+        // Find the regular method/field level any-setter
+        AnnotatedMember anySetter = beanDesc.findAnySetterAccessor();
+        if (anySetter != null) {
+            return constructAnySetter(ctxt, beanDesc, anySetter);
+        }
+        // else look for any-setter via @JsonCreator
+        if (creatorProps != null) {
+            for (SettableBeanProperty prop : creatorProps) {
```

Comment:
... I think this is where could detect duplicates instead.

And in fact should probably consider duplicates with Field/Method any-setter (if any) an error condition too (otherwise, which one to use?).
Unless there's some weird limitation wrt Records, where annotations are duplicated, so we could have Field and Constructor parameter?

Reply:
Agreed, I may have fixated too early on how multiple anysetter declaration should be handled ( thus writing that duplicate introspection in BasicDeseiraierFactory)

Reply:
Ok, it looks like there's a later point where this could be caught instead, which would make sense to me.

Reply:
As per other notes, we do need earlier check anyway.
I think ideally collection and validation would actually occur in `POJOPropertiesCollector` but that can be done with a later refactoring.